From fc7ac6a04b0f60ff5ff2af4c476ec52e68f66a0f Mon Sep 17 00:00:00 2001
From: lunatickochiya <125438787@qq.com>
Date: Sun, 23 Feb 2025 14:21:23 +0800
Subject: [PATCH 2/6] ipq807x: add support for Aliyun AP8220 mod for 24.10
 ap8220:fix wifi init issue Revert "ap8220:fix wifi issues"

This reverts commit 858b2d0aa224758a3b38753635b1afff8f52a3c1.
ap8220:fix wifi issues
ipq807x: add support for Aliyun AP8220 mod for 24.10 ipq807x: add support for Aliyun AP8220 mod for 24.10 Revert "Auxiliary commit to revert individual files from 5e3046cb221a1b7cef554a147c22307c7ff109f8"

This reverts commit b0b4368a2c22e613f5532bcf20398ffd4f3f5291.
---
 .../uboot-envtools/files/qualcommax_ipq807x   |  1 +
 package/firmware/ipq-wifi/Makefile            |  8 ++-
 .../arm64/boot/dts/qcom/ipq8071-ap8220.dts    | 56 ++++++-------------
 target/linux/qualcommax/image/ipq807x.mk      | 13 +++++
 .../ipq807x/base-files/etc/board.d/02_network |  5 ++
 .../etc/hotplug.d/firmware/11-ath11k-caldata  |  1 +
 .../etc/hotplug.d/ieee80211/11_fix_wifi_mac   |  5 ++
 .../base-files/lib/upgrade/platform.sh        |  9 +++
 8 files changed, 57 insertions(+), 41 deletions(-)

diff --git a/package/boot/uboot-envtools/files/qualcommax_ipq807x b/package/boot/uboot-envtools/files/qualcommax_ipq807x
index 2d76045c66..26f7d64c54 100644
--- a/package/boot/uboot-envtools/files/qualcommax_ipq807x
+++ b/package/boot/uboot-envtools/files/qualcommax_ipq807x
@@ -17,6 +17,7 @@ netgear,wax630)
 	[ -n "$idx" ] && \
 		ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x40000" "0x20000" "2"
 	;;
+aliyun,ap8220|\
 compex,wpq873|\
 edgecore,eap102|\
 zyxel,nbg7815)
diff --git a/package/firmware/ipq-wifi/Makefile b/package/firmware/ipq-wifi/Makefile
index 4732a14b1b..d71ee94bb3 100644
--- a/package/firmware/ipq-wifi/Makefile
+++ b/package/firmware/ipq-wifi/Makefile
@@ -6,9 +6,9 @@ PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(PROJECT_GIT)/project/firmware/qca-wireless.git
-PKG_SOURCE_DATE:=2024-10-08
-PKG_SOURCE_VERSION:=dcbab62272bf5cab2ed857bc655d240970e14f2a
-PKG_MIRROR_HASH:=e53a3872abf5e35db6baaceb56e6ffa289f8dd9b6226cf8a4d5b87b541179175
+PKG_SOURCE_DATE:=2025-02-16
+PKG_SOURCE_VERSION:=331dd0ead646f0d3fe38cb4bf83cce4b448343e8
+PKG_MIRROR_HASH:=f49b3acd56a1cf4a0d52413b060d6adbd65c2754b44cabe096fed69172febbb5
 PKG_FLAGS:=nonshared
 
 include $(INCLUDE_DIR)/package.mk
@@ -28,6 +28,7 @@ endef
 
 ALLWIFIBOARDS:= \
 	8devices_mango \
+	aliyun_ap8220 \
 	arcadyan_aw1000 \
 	asus_rt-ax89x \
 	buffalo_wxr-5950ax12 \
@@ -155,6 +156,7 @@ endef
 #   board-<devicename>.<qca4019|qca9888|qca9889|qca9984|qca99x0|ipq8074>
 
 $(eval $(call generate-ipq-wifi-package,8devices_mango,8devices Mango))
+$(eval $(call generate-ipq-wifi-package,aliyun_ap8220,Aliyun AP8220))
 $(eval $(call generate-ipq-wifi-package,arcadyan_aw1000,Arcadyan AW1000))
 $(eval $(call generate-ipq-wifi-package,asus_rt-ax89x,Asus RT-AX89X))
 $(eval $(call generate-ipq-wifi-package,buffalo_wxr-5950ax12,Buffalo WXR-5950AX12))
diff --git a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-ap8220.dts b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-ap8220.dts
index 75977afa15..f49e377002 100644
--- a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-ap8220.dts
+++ b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-ap8220.dts
@@ -8,7 +8,6 @@
 #include "ipq8074-nss.dtsi"
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
-#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Aliyun AP8220";
@@ -43,41 +42,27 @@
 		compatible = "gpio-leds";
 
 		led_power: power {
-			color = <LED_COLOR_ID_GREEN>;
-			function = LED_FUNCTION_POWER;
+			label = "green:power";
 			gpios = <&tlmm 46 GPIO_ACTIVE_HIGH>;
 		};
 
 		wlan2g {
-			color = <LED_COLOR_ID_GREEN>;
-			function = LED_FUNCTION_WLAN_2GHZ;
+			label = "green:wlan2g";
 			gpios = <&tlmm 47 GPIO_ACTIVE_HIGH>;
 			linux,default-trigger = "phy1radio";
 		};
 
 		wlan5g {
-			color = <LED_COLOR_ID_GREEN>;
-			function = LED_FUNCTION_WLAN_5GHZ;
+			label = "green:wlan5g";
 			gpios = <&tlmm 48 GPIO_ACTIVE_HIGH>;
 			linux,default-trigger = "phy0radio";
 		};
 
 		bluetooth {
-			color = <LED_COLOR_ID_GREEN>;
-			function = LED_FUNCTION_BLUETOOTH;
+			label = "green:bluetooth";
 			gpios = <&tlmm 50 GPIO_ACTIVE_HIGH>;
 		};
 	};
-
-	gpio-export {
-		compatible = "gpio-export";
-
-		ble-power {
-			gpio-export,name = "ble_power";
-			gpio-export,output = <1>;
-			gpios = <&tlmm 54 GPIO_ACTIVE_HIGH>;
-		};
-	};
 };
 
 &tlmm {
@@ -242,6 +227,10 @@
 &qpic_nand {
 	status = "okay";
 
+	partitions {
+		status = "disabled";
+	};
+
 	nand@0 {
 		reg = <0>;
 		nand-ecc-strength = <4>;
@@ -254,18 +243,8 @@
 			#size-cells = <1>;
 
 			partition@0 {
-				label = "rootfs1";
-				reg = <0x0000000 0x3000000>;
-			};
-
-			partition@3000000 {
-				label = "rootfs2";
-				reg = <0x3000000 0x3000000>;
-			};
-
-			partition@6000000 {
-				label = "usrdata";
-				reg = <0x6000000 0x2000000>;
+				label = "rootfs";
+				reg = <0x0000000 0x8000000>;
 			};
 		};
 	};
@@ -280,14 +259,12 @@
 	qca8081_24: ethernet-phy@24 {
 		compatible = "ethernet-phy-id004d.d101";
 		reg = <24>;
-		reset-deassert-us = <10000>;
 		reset-gpios = <&tlmm 33 GPIO_ACTIVE_LOW>;
 	};
 
 	qca8081_28: ethernet-phy@28 {
 		compatible = "ethernet-phy-id004d.d101";
 		reg = <28>;
-		reset-deassert-us = <10000>;
 		reset-gpios = <&tlmm 44 GPIO_ACTIVE_LOW>;
 	};
 };
@@ -295,10 +272,14 @@
 &switch {
 	status = "okay";
 
-	switch_lan_bmp = <ESS_PORT5>;
-	switch_wan_bmp = <ESS_PORT6>;
-	switch_mac_mode1 = <MAC_MODE_SGMII_PLUS>;
-	switch_mac_mode2 = <MAC_MODE_SGMII_PLUS>;
+	switch_cpu_bmp = <0x1>;  /* cpu port bitmap */
+	switch_lan_bmp = <0x20>; /* lan port bitmap */
+	switch_wan_bmp = <0x40>; /* wan port bitmap */
+	switch_mac_mode = <0xff>; /* mac mode for uniphy instance0*/
+	switch_mac_mode1 = <0xf>; /* mac mode for uniphy instance1*/
+	switch_mac_mode2 = <0xf>; /* mac mode for uniphy instance2*/
+	bm_tick_mode = <0>; /* bm tick mode */
+	tm_tick_mode = <0>; /* tm tick mode */
 
 	qcom,port_phyinfo {
 		port@5 {
@@ -320,7 +301,6 @@
 
 &dp5 {
 	status = "okay";
-	phy-mode = "sgmii";
 	phy-handle = <&qca8081_24>;
 	label = "wan";
 };
diff --git a/target/linux/qualcommax/image/ipq807x.mk b/target/linux/qualcommax/image/ipq807x.mk
index bdf7dcc82c..3cbfc4480a 100644
--- a/target/linux/qualcommax/image/ipq807x.mk
+++ b/target/linux/qualcommax/image/ipq807x.mk
@@ -33,6 +33,19 @@ define Build/wax6xx-netgear-tar
 	rm -rf $@.tmp
 endef
 
+define Device/aliyun_ap8220
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_VENDOR := Aliyun
+	DEVICE_MODEL := AP8220
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	DEVICE_DTS_CONFIG := config@ac02
+	SOC := ipq8071
+	DEVICE_PACKAGES := ipq-wifi-aliyun_ap8220
+endef
+TARGET_DEVICES += aliyun_ap8220
+
 define Device/arcadyan_aw1000
 	$(call Device/FitImage)
 	$(call Device/UbiFit)
diff --git a/target/linux/qualcommax/ipq807x/base-files/etc/board.d/02_network b/target/linux/qualcommax/ipq807x/base-files/etc/board.d/02_network
index edb1ae9557..6fcb3d4eb1 100644
--- a/target/linux/qualcommax/ipq807x/base-files/etc/board.d/02_network
+++ b/target/linux/qualcommax/ipq807x/base-files/etc/board.d/02_network
@@ -36,6 +36,7 @@ ipq807x_setup_interfaces()
 	xiaomi,ax3600)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
 		;;
+	aliyun,ap8220|\
 	edgecore,eap102|\
 	yuncore,ax880|\
 	zte,mf269)
@@ -81,6 +82,10 @@ ipq807x_setup_macs()
 	local label_mac=""
 
 	case "$board" in
+		aliyun,ap8220)
+			wan_mac=$(cat /dev/mtd12 | head -n 4 | grep "product.mac" | awk -F " " '{print $2}')
+			lan_mac=$(macaddr_add $wan_mac 1)
+		;;
 		linksys,mx4200v2|\
 		linksys,mx4300)
 			label_mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
diff --git a/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata b/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata
index 34c0a476d3..e3ec902510 100644
--- a/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata
+++ b/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata
@@ -13,6 +13,7 @@ case "$FIRMWARE" in
 		CI_UBIPART="UBI_DEV"
 		caldata_extract_ubi "Factory" 0x1000 0x20000
 		;;
+	aliyun,ap8220|\
 	arcadyan,aw1000|\
 	buffalo,wxr-5950ax12|\
 	cmcc,rm2-6|\
diff --git a/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac b/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac
index 80c07b8542..3cc120265f 100644
--- a/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac
+++ b/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac
@@ -10,6 +10,11 @@ PHYNBR=${DEVPATH##*/phy}
 board=$(board_name)
 
 case "$board" in
+	aliyun,ap8220)
+		wan_mac=$(cat /dev/mtd12 | head -n 4 | grep "product.mac" | awk -F " " '{print $2}')
+		[ "$PHYNBR" = "0" ] && macaddr_add $wan_mac 2 > /sys${DEVPATH}/macaddress
+		[ "$PHYNBR" = "1" ] && macaddr_add $wan_mac 3 > /sys${DEVPATH}/macaddress
+		;;
 	arcadyan,aw1000)
 		[ "$PHYNBR" = "0" ] && macaddr_add $(get_mac_label) 1 > /sys${DEVPATH}/macaddress
 		[ "$PHYNBR" = "1" ] && macaddr_add $(get_mac_label) 2 > /sys${DEVPATH}/macaddress
diff --git a/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh b/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh
index 0691c92184..d67306254c 100644
--- a/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh
+++ b/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh
@@ -127,6 +127,15 @@ platform_pre_upgrade() {
 
 platform_do_upgrade() {
 	case "$(board_name)" in
+	aliyun,ap8220)
+		active="$(fw_printenv -n active)"
+		if [ "$active" -eq "1" ]; then
+			CI_UBIPART="rootfs1"
+		else
+			CI_UBIPART="rootfs2"
+		fi
+		nand_do_upgrade "$1"
+		;;
 	arcadyan,aw1000|\
 	cmcc,rm2-6|\
 	compex,wpq873|\
-- 
2.34.1

